lib/Caxy/HtmlDiff/ListDiff.php

```diff
@@ -0,0 +1,366 @@
+    protected $isolatedDiffTags = array (
+        'ol' => '[[REPLACE_ORDERED_LIST]]',
+        'ul' => '[[REPLACE_UNORDERED_LIST]]',
+        'dl' => '[[REPLACE_DEFINITION_LIST]]',
+    );
+    
+    protected $listType; // holds the type of list this is ol, ul, dl
+    protected $list; // hold the old/new content of the content of the list
+    protected $childLists; // contains the old/new child lists content within this list
+    protected $textMatches; // contains the old/new text strings that match
+    protected $listsIndex; // contains the indexed start positions of each list within word string.
+    
+    /**
+     * We're using the same functions as the parent in build() to get us to the point of 
+     * manipulating the data within this class.
+     */
+    public function build()
+    {
+        // Use the parent functions to get the data we need organized.
+        $this->splitInputsToWords();
+        $this->replaceIsolatedDiffTags();
+        $this->indexNewWords();
+        // Now use the custom functions in this class to use the data and generate our diff.
+        $this->diffListContent();
+        
+        return $this->content;
+    }
+    
+    /**
+     * Calls to the actual custom functions of this class, to diff list content.
+     */
+    protected function diffListContent()
+    {
+        /* Format the list we're focusing on.
+         * There will always be one list, though passed as an array with one item.
+         * Format this to only have the list contents, outside of the array.
+         */
+        $this->formatThisListContent();
+        /* In cases where we're dealing with nested lists,
+         * make sure we use placeholders to replace the nested lists
+         */
+        $this->replaceListIsolatedDiffTags();
+        /* Build a list of matches we can reference when we diff the contents of the lists.
+         * This is needed so that we each NEW list node is matched against the best possible OLD list node/
+         * It helps us determine whether the list was added, removed, or changed.
+         */
+        $this->matchAndCompareLists();
+        /* Go through the list of matches, and diff the contents of each.
+         * Any nested lists would be sent to parent's diffList function, which creates a new listDiff class.
+         */
+        $this->diff();
+    }
+    
+    /*
+     * This function is used to remove the wrapped ul, ol, or dl characters from this list
+     * and sets the listType as ul, ol, or dl, so that we can use it later.
+     * $list is being set here as well, as an array with the old and new version of this list content.
+     */
+    protected function formatThisListContent()
+    {
+        foreach ($this->oldIsolatedDiffTags as $key => $diffTagArray) {
+            $openingTag = preg_replace("/[^A-Za-z0-9 ]/", '', $diffTagArray[0]);
+            $closingTag = preg_replace("/[^A-Za-z0-9 ]/", '', $diffTagArray[count($diffTagArray) - 1]);
```

Comment:
These tags may have additional attributes, so this preg_replace will not work in those cases -- I think you'll want to do a regex similar to the one done in HtmlDiff::isOpeningTag / HtmlDiff::isClosingTag.

You could probably create a function that returns the tag name, that you could pass in either an opening or closing tag. Something like:
`/<\/?([a-zA-Z0-9]+)[^>]*>/`
And then you can get the value of capture group.

Regex101 example: https://regex101.com/r/qQ4vN6/2